src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsServices/Helpers/ServiceHelper.cs

```diff
@@ -0,0 +1,267 @@
+    {
+        var services = ServiceController.GetServices().OrderBy(s => s.DisplayName);
+        IEnumerable<ServiceController> serviceList = new List<ServiceController>();
+
+        if (search.StartsWith(Resources.wox_plugin_service_status + ":", StringComparison.CurrentCultureIgnoreCase))
+        {
+            // allows queries like 'status:running'
+            serviceList = services.Where(s => GetLocalizedStatus(s.Status).Contains(search.Split(':')[1], StringComparison.CurrentCultureIgnoreCase));
+        }
+        else if (search.StartsWith(Resources.wox_plugin_service_startup + ":", StringComparison.CurrentCultureIgnoreCase))
+        {
+            // allows queries like 'startup:automatic'
+            serviceList = services.Where(s => GetLocalizedStartType(s.StartType, s.ServiceName).Contains(search.Split(':')[1], StringComparison.CurrentCultureIgnoreCase));
+        }
+        else
+        {
+            // To show 'starts with' results first, we split the search into two steps and then concatenating the lists.
+            var servicesStartsWith = services
+                .Where(s => s.DisplayName.StartsWith(search, StringComparison.OrdinalIgnoreCase) || s.ServiceName.StartsWith(search, StringComparison.OrdinalIgnoreCase));
+            var servicesContains = services.Except(servicesStartsWith)
+                .Where(s => s.DisplayName.Contains(search, StringComparison.OrdinalIgnoreCase) || s.ServiceName.Contains(search, StringComparison.OrdinalIgnoreCase));
+            serviceList = servicesStartsWith.Concat(servicesContains);
+        }
+
+        return serviceList.Select(s =>
+        {
+            ServiceResult serviceResult = new ServiceResult(s);
+            ServiceCommand serviceAction;
+            CommandContextItem[] moreCommands;
+            if (serviceResult.IsRunning)
+            {
+                serviceAction = new ServiceCommand(serviceResult, Action.Stop);
+                moreCommands = [
+                    new CommandContextItem(new RestartServiceCommand(serviceResult)),
+                    new CommandContextItem(new OpenServicesCommand(serviceResult)),
+                ];
+            }
+            else
+            {
+                serviceAction = new ServiceCommand(serviceResult, Action.Start);
+                moreCommands = [
+                    new CommandContextItem(new OpenServicesCommand(serviceResult)),
+                ];
+            }
+
+            return new ListItem(serviceAction)
+            {
+                Title = s.DisplayName,
+                Subtitle = ServiceHelper.GetResultSubTitle(s),
+                MoreCommands = moreCommands,
+
+                // ToolTipData = new ToolTipData(serviceResult.DisplayName, serviceResult.ServiceName),
+                // IcoPath = icoPath,
+            };
+        });
+    }
+
+    // TODO: IPublicAPI contextAPI isn't used anymore, but we need equivalent ways to show notifications and status
```

Comment:
Something like https://github.com/zadjii-msft/PowerToys/issues/118?

Reply:
Yeah exactly like that.